sanic/blueprints.py

```diff
@@ -81,6 +115,18 @@ def __repr__(self) -> str:
         )
         return f"Blueprint({args})"
 
+    def __setattr__(self, name: str, value: Any) -> None:
+        # This is a temporary compat layer so we can raise a warning until
+        # setting attributes on the app instance can be removed and deprecated
+        # with a proper implementation of __slots__
+        if name not in self.__fake_slots__:
+            warn(
+                "Setting variables on blueprint instances is deprecated and "
+                "will be removed in version 21.9. You should change your "
+                f"blueprint to use bp.ctx.{name} instead."
+            )
+        super().__setattr__(name, value)
+
```

Comment:
Should we add this to a base class (and do the same with `Application`) to map deprecated methods on one specific place? I don't know if that's actually a good idea, but I'll leave here for a thought.